_source/_posts/2022-06-01-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,604 @@
+
+Once the installation is complete, we need to fetch the External IP of the Istio Ingress Gateway. If you are using KDash, you can see it on the services tab, or you can run this command to get it using kubectl: `kubectl get svc istio-ingressgateway -n istio-system`
+
+### Install Observability tools
+
+Istio also provides addons for most of the popular monitoring and observability tools. Lets install Grafana, Prometheus, Kiali and Zipkin on our cluster. These are pre-configured to work with the telemetry data provided by Istio. Ensure you are in the folder where you installed Istio, like **istio-1.13.4**.
+
+```bash
+cd istio-$ISTIO_VERSION
+kubectl apply -f samples/addons/grafana.yaml
+kubectl apply -f samples/addons/prometheus.yaml
+kubectl apply -f samples/addons/kiali.yaml
+kubectl apply -f samples/addons/extras/zipkin.yaml
+```
+
+{% img blog/cloud-native-java-microservices-with-istio/istio-pods.png alt:"GKE Cluster with Istio pods" width:"900" %}{: .center-image }
+
+If we look at the istio-system namespace, we can see all the Istio components along with Grafana, Prometheus, Kiali, and Zipkin running. You can also see this by running `kubectl get pods -n istio-system`.
+
+## Deploy the microservice stack to GKE
+
+Our cluster is ready, and we have Istio installed. Now, we can deploy our microservice stack to the cluster. But before that, we need to create Kubernetes manifests for our deployments and services and configurations for Istio. And once again, JHipster comes to the rescue. We can use the [JDL deployment](https://www.jhipster.tech/jdl/deployments) configurations to generate Kubernetes setup for our stack with one command easily.
+
+### Create Kubernetes manifests
+
+Create a new JDL file, say **deployment.jdl**, and add the following content.
+
+```kotlin
+// will be created under 'kubernetes' folder
+deployment {
+  deploymentType kubernetes
+  appsFolders [store, invoice, notification, product]
+  dockerRepositoryName "<your-docker-repository-name>"
+  serviceDiscoveryType no
+  istio true
+  kubernetesServiceType Ingress
+  kubernetesNamespace jhipster
+  ingressDomain "<istio-ingress-gateway-external-ip>.nip.io"
+  ingressType gke
+}
+```
+
+I hope it's self-explanatory. You can refer to the JDL deployment documentation for all the available options. We have enabled Istio and set the ingress domain to the Istio Ingress Gateway's external IP that we noted earlier. Make sure to use a docker repo where you have push rights.
+
+Now run the following command from the root folder where you ran the previous `jhipster jdl` command.
+
+```bash
+jhipster jdl deployment.jdl
+```
+
+This will create a new folder, **kubernetes**, with all the required Kubernetes manifests like deployments, services, Istio virtual services, gateways, and so on for all the applications, databases, and monitoring.
```

Comment:
```suggestion
This will create a new folder, **kubernetes**, with all the required Kubernetes manifests like deployments, services, Istio virtual services, gateways, and so on, for all the applications, databases, and monitoring.
```